autopush/web/health.py

```diff
@@ -61,7 +61,9 @@ def _check_error(self, failure, name):
         cause = self._health_checks[name] = {"status": "NOT OK"}
         if failure.check(MissingTableException):
             cause["error"] = failure.value.message
-        elif failure.check(InternalServerError):  # pragma nocover
+        elif (failure.check(ClientError) and
+              failure.value.response["Error"]["Code"] ==
+              "InternalServerError"):  # pragma nocover
```

Comment:
This is becoming a reoccurring pattern. I wonder if we should create a function that just does this combined check?

Reply:
Totally, I almost created one in this diff but figured I'd keep these short/readable. Logged https://github.com/mozilla-services/autopush/issues/1142